pkg/image/options.go

```diff
@@ -52,18 +52,10 @@ func GetOptions(ctx context.Context, imageName name.Reference, insecure bool, do
 	options = append(options, remote.WithContext(ctx))
 
 	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.TLSClientConfig = &tls.Config{MinVersion: tls.VersionTLS12}
```

Comment:
With `insecure` set to true, we in the past also did not set a MinVersion. No sure if we should change this.

Reply:
My rational was that with insure, the min version should not matter. But that was a hunch. I can also rewrite it to be more like the previous style.

Reply:
Hm, just notice that we forgot to even document the `spec.output.insecure` field in both Build and BuildRun. :-( And in the proposal we have not defined it in details (but wrote that it should be documented :-) ).

Reply:
There is sadly no other way to write that without getting yet another `gosec` issue.